docs/adrs/004-SplitAndAggregate.md

```diff
@@ -0,0 +1,139 @@
+# Replace KBParallels with another solution to avoid Deadlocks
+
+Date: 2021-09-22
+
+[Related ADR](https://github.com/kbase/execution_engine2/blob/develop/docs/adrs/rework-batch-analysis-architecture.md)
+
+## Note
+This ADR is more of a place to keep the current discussions we had at https://docs.google.com/document/d/1AWjayMoqCoGkpO9-tjXxEvO40yYnFtcECbdne5vTURo
+Rather than to make a decision. There is still more planning, scoping and testing involved before we can fully design this system.
+
+Still to be determined (not in scope of this ADR):
+* UI and how it relates to the bulk execution
+* XSV Analysis and how it relates to the bulk execution
+
+
+## Intro
+Sometimes a calculation requires too many resources from one node (walltime, memory, disk), so the calculation gets spread across multiple machines.
+The final step of the app that uses KBParallels is to create a report. This step may use results from all of the computed jobs to create the final report.
+In order to do this, some apps use a mechanism called KBParallel. The apps are listed at [DATAUP-111](https://kbase-jira.atlassian.net/browse/DATAUP-111)
```

Comment:
This list only includes in scope apps for BU, whereas this ADR is really about if we need to support apps outside that list that do job batching

Reply:
:+1: